envoy/router/context.h

```diff
@@ -21,6 +22,11 @@ class Context {
    * @return a struct containing StatNames for virtual cluster stats.
    */
   virtual const VirtualClusterStatNames& virtualClusterStatNames() const PURE;
+
+  /**
+   * @return a pointer to the default generic connection pool factory.
+   */
+  virtual GenericConnPoolFactory* genericConnPoolFactory() const PURE;
```

Comment:
per Envoy convention let's make this return a reference. This also can help seed static analysis tools (which I'm starting to think about) with an invariant that this can't be null.